src/main/java/org/sciborgs1155/robot/Hopper/Hopper.java

```diff
@@ -0,0 +1,54 @@
+package org.sciborgs1155.robot.Hopper;
+
+import static org.sciborgs1155.robot.Hopper.HopperConstants.*;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import monologue.Logged;
+import monologue.Monologue.LogBoth;
+import org.sciborgs1155.robot.Robot;
+
+public class Hopper extends SubsystemBase implements Logged {
+
+  final PIDController pid = new PIDController(kp, ki, kd);
+  public final HopperIO hopper = Robot.isReal() ? new RealHopper() : new SimHopper();
+
+  @LogBoth public double targetSpeed = pid.getSetpoint();
+
+  @LogBoth
+  public double speed() {
+    return hopper.getSpeed();
+  }
+
+  @LogBoth private boolean isAtTarget = pid.atSetpoint();
```

Comment:
currently, this is a private field. You set it once at the beginning to be pid.atSetpoint(), but it doesn't update, so assuming your starting setpoint is 0, it'll start out as true and then just continue to be true the entire time. If you instead make this a method, (so like private boolean isAtTarget() { return pid.atSetpoint(); }) every tick it would log the new value from that method, so it'd update.